### PR TITLE
Document and test custom analyzer `position_offset_gap`

### PR DESCRIPTION
**Pull request by mattweber**
Document and add a quick test for setting a `position_offset_gap` on a custom analyzer.  This functionality was added way back in 2012 in #1812 but I just recently discovered it and feel that it should not go undocumented.

### CODE REVIEW

**Comment by clintongormley** [comment]
++ I've always wondered what this option did :)

**Comment by s1monw** [comment]
@rjernst can you review this please

File: docs/reference/analysis/analyzers/custom-analyzer.asciidoc [hunk 1]
```diff
@@ -18,6 +18,9 @@ filters.
 
 |`char_filter` |An optional list of logical / registered name of char
 filters.
+
```
Comment (rjernst) [inline docs/reference/analysis/analyzers/custom-analyzer.asciidoc]: Can change to not rely on someone knowing what a  "gap" is here?
```
An optional number of positions to increment between each value for a field.
```

Comment (mattweber) [inline docs/reference/analysis/analyzers/custom-analyzer.asciidoc]: Done.

**Comment by rjernst** [comment]
Thanks @mattweber!

**Pull request merged**